include/inviwo/core/datastructures/geometry/typedmesh.h

```diff
@@ -0,0 +1,489 @@
+    void addVertexImpl(T &t) {
+        using BT = typename std::tuple_element<I, std::tuple<BufferTraits...>>::type;
+        BT::getTypedEditableRAMRepresentation()->add(t);
+    }
+
+    template <unsigned I, typename T, typename... ARGS>
+    void addVertexImpl(T &t, ARGS... args) {
+        using BT = typename std::tuple_element<I, std::tuple<BufferTraits...>>::type;
+        BT::getTypedEditableRAMRepresentation()->add(t);
+        addVertexImpl<I + 1>(args...);
+    }
+
+    template <unsigned I, typename T>
+    void setVertiesImpl(size_t index, T &t) {
+        using BT = typename std::tuple_element<I, std::tuple<BufferTraits...>>::type;
+        BT::getTypedDataContainer().at(index) = t;
+    }
+
+    template <unsigned I, typename T, typename... ARGS>
+    void setVertiesImpl(size_t index, T &t, ARGS... args) {
+        using BT = typename std::tuple_element<I, std::tuple<BufferTraits...>>::type;
+        BT::getTypedDataContainer().at(index) = t;
+        setVertiesImpl<I + 1>(index, args...);
+    }
+
+    template <unsigned I>
+    void addVerticesImpl(const std::vector<Vertex> &vertices) {}  // sink
+
+    template <unsigned I, typename T, typename... ARGS>
+    void addVerticesImpl(const std::vector<Vertex> &vertices) {
+
+        auto &vec = getTypedDataContainer<T>();
+
+        auto neededSize = vertices.size() + vec.size();
+        if (vec.capacity() < neededSize) {
+            vec.reserve(neededSize);
+        }
+        for (auto &v : vertices) {
+            vec.push_back(std::get<I>(v));
+        }
+
+        addVerticesImpl<I + 1, ARGS...>(vertices);
+    }
+
+    template <unsigned I>
+    void copyConstrHelper() {}  // sink
+
+    template <unsigned I, typename T, typename... ARGS>
+    void copyConstrHelper() {
+        T::buffer_ = std::static_pointer_cast<Buffer<typename T::type>>(Mesh::buffers_[I].second);
+        copyConstrHelper<I + 1, ARGS...>();
+    }
+};
+
+/**
+ * \ingroup typedmesh
+ * Type definition of a TypedMesh useful for Spheres, consists of a vec3-buffer for position, a
+ * float-buffer for radii and vec4 for colors.
+ */
+using SphereMesh = TypedMesh<buffertraits::PositionsBuffer, buffertraits::RadiiBuffer,
+                                 buffertraits::ColorsBuffer>;
+
+/**
+ * \ingroup typedmesh
+ * Type definition of a TypedMesh having only positions(vec3) and colors(vec4).
+ * Example usage:
+ * \snippet modules/base/algorithm/meshutils.cpp Using Colored Mesh
+ */
+using ColoredMesh = TypedMesh<buffertraits::PositionsBuffer, buffertraits::ColorsBuffer>;
+
+
+/**
+ * \ingroup typedmesh
+ * Type definition of a TypedMesh having positions(vec3), normals(vec3), texture
+ * coordinates(vec3) and colors(vec4). Example usage:
+ */
+using BasicMesh = TypedMesh<buffertraits::PositionsBuffer, buffertraits::NormalBuffer,
+                                buffertraits::TexcoordBuffer, buffertraits::ColorsBuffer>;
+
+
+
+
+using NewSimpleMesh = TypedMesh<buffertraits::PositionsBuffer,
```

Comment:
`NewSimpleMesh` :)

Reply:
not entirely sure whether this is a good name though. In particular since it will be used in a lot of places...

Reply:
This was for testing